### ISSUE TITLE
#include preprocessing directive not working

### ISSUE THREAD

**Issue by JonathanWheelerL3** [jmmorato/openddsharp#99]
I've been making some IDL files to test some things and have created "CommonDefinitions.idl". For simplification, the code of that file is

```
typedef sequence<string> StringList;
```

I then made "TestModule.idl" and put the following code in it:

```
#include <CommonDefinitions.idl>

module TestModule{

	@topic
	struct TestMessage {
		string Source;
		StringList Messages;
	};

};
```

Upon rebuilding the solution, I get errors like `'StringList': is not a member of 'OpenDDSharp'` in "TestModuleTypeSupport.h".

I have tried changing the include directive to `#include "CommonDefinitions.idl"` which gives the same error.

Replacing `#include ...` with `typedef sequence<string> StringList;` and rebuilding will work correctly.

Does the `#include` directive not work in OpenDDSharp?

For reference, here is the test project layout.
![image](https://user-images.githubusercontent.com/96193702/148820643-550403c1-ec89-4567-8bbb-1d7c0d1ca74b.png)

**Comment by jmmorato** [comment]
it should work but...
could you please zip and attach your project for testing?

**Comment by JonathanWheelerL3** [comment]
Here you go.
[FxIcd.zip](https://github.com/jmmorato/openddsharp/files/7848181/FxIcd.zip)

**Comment by jmmorato** [comment]
Sorry for my late response, I'm in a bussiness travel with limited access to my personal computer.

I can confirm there is a bug in the #include directive. For the moment, the only workaround I could provide is to manually modify the `TestModuleTypeSupport.h` and manually include the directive, you would need to remove the `SUPPRESS_LNK4248_CORBA` from that file as well.

```
#include <dds/DCPS/Service_Participant.h>
#include "TestModuleTypeSupportImpl.h" 
#include "CommonDefinitionsTypeSupport.h"
#pragma managed

#include <vcclr.h>
#include <msclr/marshal.h>
//#include "LNK4248.h"

using namespace System::Collections::Generic;

// Incomplete types generate LNK4248 warnings when compiled for .NET
//SUPPRESS_LNK4248_CORB
```

After that build the project again. If you Rebuild the project or modify your IDL file, the code will be regenerated and your manual modifications will be lost and you will need to add it again.

I will ensure the bug is fixed for next versions of the standard wrapper (CPP/CLI version is frozen and no new versions are planned)

**Comment by jmmorato** [comment]
Fix released in latest version 3.250.2
